src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
+  }
```

Comment:
[CODE PERFOMANCE] - Instead of using `indexOf` and `splice` in a loop to remove dead animals, prefer using `filter` to avoid nesting loops and improve performance. Consider filtering `Animal.alive` directly in the `bite` method after decreasing the health or create a separate method to handle the removal of dead animals from the `Animal.alive` array.